datadog/client.go

```diff
@@ -1,47 +1,131 @@
 package datadog
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
 
 	"github.com/BESTSELLER/dependabot-circleci/config"
-	"github.com/DataDog/datadog-go/statsd"
 	"github.com/rs/zerolog/log"
 )
 
-var client *statsd.Client
-var metricPrefix = "dependabot_circleci"
+type DataDog struct {
+	Series []Series `json:"series"`
+}
 
-// CreateClient creates a statsd client
-func CreateClient() (err error) {
-	client, err = statsd.New(config.EnvVars.DDAddress)
-	if err != nil {
-		return err
-	}
-	return nil
+type Series struct {
+	Metric   string    `json:"metric"`
+	Points   [][]int64 `json:"points"`
+	Tags     []string  `json:"tags"`
+	Type     string    `json:"type"`
+	Host     string    `json:"host"`
+	Interval int64     `json:"interval"`
 }
 
+var metricPrefix = "dependabot_circleci"
+
 // IncrementCount incrementes a counter based on the input
 func IncrementCount(metricName string, org string) {
-	err := client.Incr(
-		fmt.Sprintf("%s.%s", metricPrefix, metricName),
-		[]string{
-			"organistation:" + org,
-		},
-		1)
+	metric := metricPrefix + "." + metricName
+	err := postDataDogMetric(metric, 1, "count", []string{"organistation:" + org})
 	if err != nil {
 		log.Debug().Err(err).Msgf("could increment datadog counter %s", metricName)
 	}
 }
 
 // Gauge incrementes a counter based on the input
 func Gauge(metricName string, value float64, tags []string) {
-	err := client.Gauge(
-		fmt.Sprintf("%s.%s", metricPrefix, metricName),
-		value,
-		tags,
-		1,
-	)
+	metric := metricPrefix + "." + metricName
+	err := postDataDogMetric(metric, int64(value), "gauge", tags)
 	if err != nil {
 		log.Debug().Err(err).Msgf("could send gauge to datadog %s", metricName)
 	}
+
+}
+
+// func getDataDogConfig() (apiKey string, err error) {
```

Comment:
Is this needed or should it just be removed ?

Reply:
removed